src/blockdata/block.rs

```diff
@@ -176,7 +176,10 @@ impl Block {
 
     /// check if merkle root of header matches merkle root of the transaction list
     pub fn check_merkle_root (&self) -> bool {
-        self.header.merkle_root == self.merkle_root()
+        match self.merkle_root() {
+            Some(merkle_root) => self.header.merkle_root == merkle_root,
+            None => false,
+        }
```

Comment:
I played with the combinator here as suggested and also on the call to `self.witness_root()`. It wasn't clearly better either way to me so I ended going with the code as it is since seemed _slightly_ easier to read.